joomla/joomla-cms
# Converting /includes/framework.php to a Class & refactoring

Converting /includes/framework.php from linear php file to use class structure. Simplifying the two separate files `framework.php` in **site** and **administrator** each, using inheritance.

Introduce constant `JPATH_ADMINISTRATOR_DIR` to eliminate hardcoding of the backend directory name everywhere. Not used anywhere yet, but strongly intended to.

**TEST INSTRUCTION**
Download Joomla with this patch. You should be able to:
1. Open installation page at first launch and until installation is done.
2. Remove "Installation" directory without installing, it should show error message "... not found... exiting."
3. After successful installation it should not allow site / administrator navigation unless "Installation" directory is removed.
4. After removal of "installation" directory it should open site and administrator as usual with no difference.
